resources/service.rb

```diff
@@ -40,25 +40,50 @@
           default: true,
           description: 'Perform configuration file test before performing service action'
 
+property :config_test_fail_action, Symbol,
+          equal_to: %i(raise log),
+          default: :raise,
+          description: 'Action to perform upon configuration test failure.'
+
 action_class do
   def do_service_action(resource_action)
     with_run_context(:root) do
-      begin
-        edit_resource(:execute, "Run pre service #{resource_action} #{dhcpd_service_name(new_resource.ip_version)} configuration test.") do
-          command dhcpd_config_test_command(new_resource.ip_version, new_resource.config_file)
-          only_if { new_resource.config_test && %i(start restart reload).include?(resource_action) && ::File.exist?(new_resource.config_file) }
-          user dhcpd_user
+      if %i(start restart reload).include?(resource_action)
+        edit_resource(:ruby_block, "Run pre #{new_resource.service_name} #{resource_action} configuration test") do
```

Comment:
I tried running this with `enforce_idempotency: true` and ran into the following issue with this resource:
```
Resources updated in a second chef-client run:
- execute[testing - enable ipv6]
- ruby_block[Run pre dhcpd.service start configuration test]
- ruby_block[Run pre dhcpd6.service start configuration test]
```
Is there a way to ensure this doesn't get run on every converge?

Reply:
This was refactored like this based off comments from @lamont-granquist in the slack to remove the notifications aspect of how it was previous. I did run into various edge cases/bugs where I didn't get the response you'd expect and Chef would still restart onto a bad config.

I'll have a look into it to see if there's something else that can be done but we may have to live with it.

Reply:
Unless you know of a way to cleanly access the Chef service provider to check if the service is running or not (which I'm not sure is right to do as it goes against chef being declarative), then I don't think this is worth going any further with just to remove a 'secondary' run. It's only the start action that needs it so on balance I'm of the opinion it's acceptable to leave it how it is.

Reply:
Yeah, let's table this to figure out later.